lib/bolt/shell/powershell.rb

```diff
@@ -11,6 +11,13 @@ class Powershell < Shell
       def initialize(target, conn)
         super
 
+        if target.options['run-as']
```

Comment:
Ok, it turns out for transports that don't support run-as (so Winrm and even the `local` transport if we're running on Windows) we filter out all unsupported configuration options, so if `run-as` is set and using a transport that doesn't support it the configuration will never even get here to raise the warning. I think you can just remove this to avoid confusion and we can merge as is and have a separate ticket for deciding how we want to message this to users.